sdk/spring/spring-sdk-client.yml

```diff
@@ -0,0 +1,296 @@
+          - template: ../../eng/pipelines/templates/steps/generate-project-list.yml
+            parameters:
+              Artifacts: ${{ parameters.Artifacts }}
+              AdditionalModules: ${{ parameters.AdditionalModules }}
+              JobType: 'maven_dependency'
+              SDKType: ${{parameters.SDKType}}
+          - task: PythonScript@0
+            displayName: 'Generate directories variable for sparse checkout'
+            inputs:
+              scriptPath: 'eng/scripts/generate_from_source_pom.py'
+              arguments: '--set-pipeline-variable CheckoutDirectories --project-list $(ProjectList)'
+              workingDirectory: '$(System.DefaultWorkingDirectory)'
+          - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+            parameters:
+              Paths: $(CheckoutDirectories)
+              SkipDefaultCheckout: true
+          - template: ../../eng/pipelines/templates/steps/install-reporting-tools.yml
+          - script: |
+              pip install in-place & pip install termcolor
+            displayName: 'python module install'
+          - script: |
+              python ./sdk/spring/scripts/ci_update_versions.py --sbv ${{parameters.NewSpringBootVersion}}
+            displayName: 'update spring boot versions'
+          - script: |
+              python ./sdk/spring/scripts/ci_insert_dependency_management_in_pom_files.py -b ${{parameters.NewSpringBootVersion}} -c ${{parameters.NewSpringCloudVersion}}
+            displayName: 'insert dependency management'
+          - script: |
+              mvn -f sdk/spring/pom.xml -P dev --batch-mode --fail-at-end $(WagonOptions) -T 2C -DskipTests -Dgpg.skip -DtrimStackTrace=false -Dmaven.javadoc.skip=true -Dcodesnippet.skip=true -Dspotbugs.skip=true -Dcheckstyle.skip=true -Drevapi.skip=true -Djacoco.skip=true -pl $(ProjectList) -am install
+            displayName: 'maven_install'
+          - script: |
+              cd ./sdk/spring && mvn dependency:tree>tree.txt
+            displayName: 'maven_dependency'
+          - script: |
+              python ./sdk/spring/scripts/ci_read_dptree.py
+            displayName: '${{parameters.NewSpringBootVersion}}_dependencyTree_output'
+
+  - stage: TestLastSpringBootVersion
+    jobs:
+      - template: /eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml
+        parameters:
+          JobTemplatePath: /eng/pipelines/templates/jobs/ci.tests.yml
+          CloudConfig:
+            Cloud: Public
+          MatrixConfigs:
+            - ${{ each config in parameters.MatrixConfigs }}:
+                - ${{ config }}
+            - ${{ each config in parameters.AdditionalMatrixConfigs }}:
+                - ${{ config }}
+          MatrixFilters:
+            - ${{ each filter in parameters.MatrixFilters }}:
+                - ${{ filter }}
+            # Skip TestFromSource jobs for SDKType data
+            - ${{ if eq(parameters.SDKType, 'data') }}:
```

Comment:
There are some duplicates in these two stages, can we extract the common template for them?